src/extension.ts

```diff
@@ -168,6 +195,34 @@ async function stopClients(folders = folderNames()) {
 	return Promise.all(promises);
 }
 
+// Command interface
+
+function sendRequest(clientName: string, command: string, args?: any) {
```

Comment:
early feedback: arguments is a list of any `any[]`. I was keeping my backend simple and just assuming a flat/simple structure similar to a cli invokation

`rootmodule fileInFocus` which would be an ExecuteCommandParams of 
```
{
    command: "rootmodule",
    arguments: ["theFileURIInFocus"]
}
```
I am unsure of if we can have our arguments be complex like
```
{
    command: "rootmodule",
    arguments: [
        {
            file: "fileURIInFocus"
        }
    ]
}
```
Let me know if you prefer a complex request object or keeping it "flat". I imagine some commands such as a future `validate` we will be keeping flat/parity with the actual terraform cli

Reply:
There is certainly a case to be made for the complex structure as we can define types on the Golang side to marshal/unmarshal